server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/MessagesRoutes.java

```diff
@@ -50,21 +54,42 @@
 @Path("/messages")
 @Produces("application/json")
 public class MessagesRoutes implements Routes {
+    public static class MessageReIndexingTaskRegistration extends TaskFromRequestRegistry.TaskRegistration {
+        @Inject
+        public MessageReIndexingTaskRegistration(MessageIdReIndexer reIndexer, MessageId.Factory messageIdFactory) {
+            super(MailboxesRoutes.RE_INDEX, request -> reIndexer.reIndex(extractMessageId(messageIdFactory, request)));
+        }
+    }
+
+    private static MessageId extractMessageId(MessageId.Factory messageIdFactory, Request request) {
+        try {
+            return messageIdFactory.fromString(request.params(MESSAGE_ID_PARAM));
+        } catch (Exception e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Error while parsing 'messageId'")
+                .cause(e)
+                .haltError();
+        }
+    }
+
     private static final String MESSAGE_ID_PARAM = ":messageId";
     private static final String BASE_PATH = "/messages";
     private static final String MESSAGE_PATH = BASE_PATH + "/" + MESSAGE_ID_PARAM;
+    public static final String ONE_MESSAGE_TASKS = "oneMessageTasks";
```

Comment:
s/public/private ?

Reply:
Constant used in other files for bindings, so public